orangecontrib/text/preprocess.py

```diff
@@ -1,94 +1,298 @@
-        :type trans: :class: `orangecontrib.text.preprocess.Lemmatizer`
-            or :class: `orangecontrib.text.preprocess.Stemmer`
-        :return: :class: `orangecontrib.text.preprocess.Preprocessor`
+        Args:
+            lowercase (Optional[bool]): If set, transform the tokens to lower
+                case, before returning them. Defaults to False.
+            stop_words (Optional[str, list]) : Determines whether and what
+                stop words should be removed. Can remove default nltk stop
+                words for the English language or stop words provided in a
+                custom list. Defaults to None.
+            min_df (Optional[int, float]): Tokens that appear in exactly or
+                less than 'min_df' documents, will be removed. Can be
+                specified either as an actual count of documents or a
+                proportion of the corpus. Assigning this parameter when
+                pre-processing a single document, will have no effect.
+                Defaults to None.
+            max_df (Optional[int, float]): Tokens that appear in exactly or
+                more than 'max_df' documents, will be removed. Can be
+                specified either as an actual count of documents or a
+                proportion of the corpus. Assigning this parameter when
+                pre-processing a single document, will have no effect.
+                Defaults to None.
+            transformation (Optional[`orangecontrib.text
+                .preprocess.Stemmatizer`]): Name of the morphological
+                transformation method to be performed on the tokens.
+                Defaults to None.
+            use_twitter_tokenizer (Optional[bool]): Determines the use of
+            either the Twitter or default word tokenizer. Defaults to False.
+
+        Returns:
+            list: Holds a list of lists, that correspond to documents in the
+                input corpus.
+
+        Raises:
+            ValueError: If the stop words or transformation arguments are not
+                of expected values/types.
         """
-        # TODO Needs more elaborate check on list contents.
-        if not (stop_words == 'english' or isinstance(stop_words, list) or stop_words is None):
-            raise ValueError("The stop words parameter should be either \'english\', a list or None.")
-        if isinstance(trans, Stemmatizer):  # hack, since scikit does not obey lowercase when preprocesor given
-            trans.lowercase = lowercase
-        self.stop_words = stop_words
-        # TODO does including punctuation even work?
-        self.incl_punct = incl_punct
+        # Tokenizer.
+        self.tokenizer = word_tokenize
+        if use_twitter_tokenizer:
+            self.tokenizer = None   # TODO: Change when twitter is available.
+
+        # Lowercase.
         self.lowercase = lowercase
-        self.transformation = trans
+
+        # Stop words.
+        if stop_words == 'english':
```

Comment:
Why restrict ourselves to `english`? NLTK supports many languages, so for example `stop_words='german'` could work for free. All we need is check if `stop_words` is among the supported languages. At least those also work: `['danish', 'dutch', 'english', 'finnish', 'french', 'german', 'hungarian', 'italian', 'norwegian', 'portuguese', 'russian', 'spanish', 'swedish', 'turkish']` but check for complete list.